R/hms.R

```diff
@@ -1,40 +1,46 @@
 #' objects and returning a logical value; it is evaluated after every metaepoch and
-#' determines whether whole computation should be stopped.
-#' @param lsc - local stopping condition - function taking a deme and a list of MetaepochSmapshot
+#' determines whether whole computation should be stopped. See \code{\link{gsc_metaepochs_count}} for more details.
+#' @param lsc local stopping condition - function taking a deme and a list of MetaepochSmapshot
 #' objects representing previous metaepochs; it is run on every deme after it has run a metaepoch
-#' and determines whether that deme will remain active.
-#' @param sc - sprouting condition - function taking 3 arguments: an individual, a tree level
+#' and determines whether that deme will remain active. See \code{\link{lsc_max_fitness_evaluations}} for more details.
+#' @param sc sprouting condition - function taking 3 arguments: an individual, a tree level
 #' and a list of Deme objects; it determines whether the given individual can sprout a new deme
-#' on the given level.
-#' @param create_population - function taking 6 parameters: mean, lower, upper, population_size,
+#' on the given level. See \code{\link{sc_max_metric}} for more details.
+#' @param create_population function taking 6 parameters: mean, lower, upper, population_size,
 #' tree_level, sigma that returns a population for a Deme object to be created on the given
-#'  tree level.
-#' @param suggestions - matrix of individuals for the initial population of the root
-#' @param with_gradient_method - logical determining whether a gradient method should be run
+#' tree level.
+#' @param suggestions matrix of individuals for the initial population of the root
+#' @param with_gradient_method logical determining whether a gradient method should be run
 #' for all leaves at the end of the computation to refine their best solutions.
-#' @param gradient_method_args - list of parameters that are passed to the gradient method
-#' @param run_gradient_method - function - returns list with named fields: solution, population, value
-#' @param monitor_level - string - one of {'none', 'basic', 'basic_tree', 'verbose_tree'}.
-#' @param parallel - logical - \code{TRUE} when run_metaepoch runs in parallel.
-#' @param minimize - logical - \code{TRUE} when fitness shall be minimized.
+#' @param gradient_method_args list of parameters that are passed to the gradient method
+#' @param run_gradient_method function - returns list with named fields: solution, population, value
+#' @param monitor_level string - one of {'none', 'basic', 'basic_tree', 'verbose_tree'}.
+#' @param parallel logical - \code{TRUE} when run_metaepoch runs in parallel.
+#' @param minimize logical - \code{TRUE} when fitness shall be minimized.
 #'
-#' @return numeric best solution
+#' @return Returns an object of class hms.
 #' @export
+#'
+#' @examples
+#' f <- function(x) x
+#' result <- hms(fitness = f, lower = -5, upper = 5)
```

Comment:
W sekcji @examples można dodawać komentarze tak, jak normalnie w kodzie. Można by tutaj dodać komentarz, który wytłumaczy gdzie można znaleźć bardziej zaawansowane przykłady użycia.

Chodzi mi o coś takiego jak [tutaj](https://combine-australia.github.io/r-pkg-dev/documenting-functions.html) to "Row sizes must be..."
```R
#' @examples
#' ...
#' # Row sizes must be compatible when column-binding
#' try(bind_cols(tibble(x = 1:3), tibble(y = 1:2)))
```

Reply:
Ok, ma to sens. Dodam taki komentarz jak zostanie stworzona vignette'a.

Reply:
Chyba że masz pomysł gdzie można znaleźć takie przykłady? W testach?

Reply:
Ta vignette'a to będzie pewnie dobre miejsce. Niech narazie tak zostanie